ui/src/dashboards/components/dashboard_index/Table.tsx

```diff
@@ -0,0 +1,166 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import {withRouter, WithRouterProps} from 'react-router'
+import _ from 'lodash'
+
+// Components
+import {
+  Button,
+  ComponentColor,
+  IconFont,
+  ComponentSize,
+  EmptyState,
+  IndexList,
+  Alignment,
+} from 'src/clockface'
+import TableRows from 'src/dashboards/components/dashboard_index/TableRows'
+import TableHeader from 'src/dashboards/components/dashboard_index/TableHeader'
+import SortingHat from 'src/shared/components/sorting_hat/SortingHat'
+
+// Types
+import {Sort} from 'src/clockface'
+import {IndexHeaderCellProps} from 'src/clockface/components/index_views/IndexListHeaderCell'
+import {Dashboard} from 'src/types/v2'
+
+interface Props {
+  searchTerm: string
+  dashboards: Dashboard[]
+  defaultDashboardLink: string
+  onDeleteDashboard: (dashboard: Dashboard) => void
+  onCreateDashboard: () => void
+  onCloneDashboard: (dashboard: Dashboard) => void
+  onExportDashboard: (dashboard: Dashboard) => void
+  onSetDefaultDashboard: (dashboardLink: string) => void
+}
+
+type Column = IndexHeaderCellProps
+
+interface State {
+  columns: Column[]
+}
+
+class DashboardsTable extends PureComponent<Props & WithRouterProps, State> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      columns: this.columns,
+    }
+  }
+
+  public render() {
+    const {
+      dashboards,
+      onSetDefaultDashboard,
+      defaultDashboardLink,
+      onExportDashboard,
+      onCloneDashboard,
+      onDeleteDashboard,
+    } = this.props
+
+    return (
+      <IndexList>
+        <TableHeader
+          columns={this.state.columns}
+          onClickColumn={this.handleClickColumn}
+        />
+        <IndexList.Body emptyState={this.emptyState} columnCount={5}>
+          <SortingHat<Dashboard>
+            list={dashboards}
+            sortKeys={this.sortKeys}
+            directions={this.directions}
+          >
+            {ds => (
+              <TableRows
+                dashboards={ds}
+                onCloneDashboard={onCloneDashboard}
+                onExportDashboard={onExportDashboard}
+                onDeleteDashboard={onDeleteDashboard}
+                defaultDashboardLink={defaultDashboardLink}
+                onSetDefaultDashboard={onSetDefaultDashboard}
+              />
+            )}
+          </SortingHat>
+        </IndexList.Body>
+      </IndexList>
+    )
+  }
+
+  private get sortKeys(): string[] {
+    return this.state.columns.map(c => c.sortKey)
+  }
+
+  private get directions(): Sort[] {
+    return this.state.columns.map(c => c.sort)
+  }
+
+  private handleClickColumn = (nextSort: Sort, sortKey: string) => {
+    const columns = this.state.columns.map(h => {
+      if (h.sortKey === sortKey) {
+        return {...h, sort: nextSort}
+      }
+
+      return h
+    })
+
+    this.setState({columns})
+  }
+
+  private get columns(): Column[] {
```

Comment:
_nit_

```suggestion
  private get defaultColumns(): Column[] {
```

maybe? I was confused that `columns` referred both to a getter and to something in state